### PR TITLE
Add a notation for the empty type.

### PR DESCRIPTION
**Pull request by arthuraa**
**Kind:** feature

- [x] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).
- [x] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

File: doc/changelog/06-ssreflect/10932-void-type-ssr.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+- Add a `void` notation for the standard library empty type (`Empty_set`)
```
Comment (Zimmi48) [inline doc/changelog/06-ssreflect/10932-void-type-ssr.rst]: This further marks the type of quoted stuff to belong to the Coq realm.

```suggestion
- Add a :g:`void` notation for the standard library empty type (:g:`Empty_set`)
```

**Review by Zimmi48 — APPROVED** [review]
Changelog is fine (if SSR maintainers want this).

**Comment by cpitclaudel** [comment]
Is `void` really the right name, though? I thought void was essentially the unit type.

**Comment by arthuraa** [comment]
@cpitclaudel In the C world, definitely. But many functional languages have adopted `void` as a name for the empty type (e.g. https://hackage.haskell.org/package/void-0.6.1/docs/Data-Void.html). And there was already a comment in ssr's `eqtype` referring to a `void` type that was different from `unit`.

**Comment by amahboubi** [comment]
@Zimmi48, yes we are fine with this.

**Comment by Zimmi48** [comment]
@amahboubi I take this as a positive review of the PR. Do you wish me to merge it?

**Review by amahboubi — APPROVED** [review]

**Comment by amahboubi** [comment]
@Zimmi48 Yes, please proceed.

**Pull request merged**